### DISCUSSION TITLE
ont-shot-vfl SSL

### DISCUSSION THREAD

**Discussion by wen08** [NVIDIA/NVFlare#2476] [Q&A] [answered]
### Python version (`python3 -V`)

3.8

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

python3 -m pip list | grep "nvflare"

### Operating system

windows

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [ ] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

In the one-shot-vfl project, the paper mentions the use of Local SSL, but I couldn't find where it is specifically implemented. If semi-supervised learning (SSL) is used, it should be reflected in the _local_train function in the cifar10_leamer_oneshotVFL.py file. However, I haven't found it there. Can someone please point out where the SSL-related code is located?

**Comment by chesterxgchen** [comment]
@holgerroth  do you help here ?

**Reply by holgerroth** [reply, marked as answer]
@wen08 , you're right. The semi-supervised learning approach wasn't implemented in this example. Only the one-shot VFL on overlapping samples as baseline. For the paper, we used this repo to implement semi-supervised learning https://github.com/kekmodel/FixMatch-pytorch?tab=readme-ov-file.

**Reply by wen08** [reply]
Thank you very much
